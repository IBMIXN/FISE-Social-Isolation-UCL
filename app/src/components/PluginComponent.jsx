import React from "react";
import { Box } from "@chakra-ui/core";
import theme from "../theme";
import Frame from "react-frame-component";
import Plugin from "./Plugin";

const PluginComponent = () => {
  const iframeId = "plugin-iframe";

  const iframeStyle = {
    width: "100%",
    height: "100%",
    border: "none",
    background: "none transparent",
  };

  return (
    <Box
      pos="absolute"
      top="2%"
      bottom="2%"
      right="4%"
      left="4%"
      bg={theme.colors.whiteGlass}
      pr="1rem"
      pb="1rem"
      pt="0.5rem"
      pl="0.5rem"
      roundedBottom="5px"
      roundedTop="5px"
    >
      <Frame
        title="iframeTest"
        frameborder="0"
        id={iframeId}
        objectFit="scale-down"
        style={iframeStyle}
      >
        <Plugin iframeId={iframeId}></Plugin>
      </Frame>
    </Box>
  );
};

export default PluginComponent;
