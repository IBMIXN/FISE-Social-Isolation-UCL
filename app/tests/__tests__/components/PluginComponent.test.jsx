import React from 'react';
import ReactDOM from 'react-dom';
import PluginComponent from '../../../src/components/PluginComponent';

import Enzyme, { shallow, render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() })

it('render plugin component unit test', () => {
  const wrapper = shallow(<PluginComponent/>)
  
  expect(toJson(wrapper)).toMatchSnapshot();
});

// it('integration test', () => {
//   const wrapper = mount(<PluginComponent />)

//   expect(toJson(wrapper)).toMatchSnapshot();
// });