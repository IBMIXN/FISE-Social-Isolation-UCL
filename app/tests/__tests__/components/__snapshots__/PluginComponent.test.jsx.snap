// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render plugin component unit test 1`] = `
<iframe
  frameborder="0"
  id="iframe"
  objectFit="scale-down"
  scrolling="auto"
  src="undefined/plugins.html"
  style={
    Object {
      "background": "#FFFFFF",
      "border": "none",
      "height": "100%",
      "width": "100%",
    }
  }
  title="iframeTest"
/>
`;
